tests/test_stream_app.py

```diff
@@ -110,36 +86,213 @@ def test_filter_by_value_saved_for_next_run(filter_by, record_attr, settings):
-            ) % settings.APP_KEY
-    context = SimpleNamespace(client_context=None)
+        '[{"records": [{"asset_id": 0, "timestamp": 0}], "metadata": {"app_stream_id": 0, '
+        '"apps": {"%s": {"app_connection_id": 0}}}}]'
+    ) % SETTINGS.APP_KEY
+
+    corva = Corva(SimpleNamespace(client_context=None))
+
+    if raises:
+        exc = pytest.raises(ValueError, corva.stream, stream_app, event, **kwargs)
+        assert (
+            'filter_by_timestamp and filter_by_depth can\'t be set to True together.'
+            in str(exc.value)
+        )
+
+
+@pytest.mark.parametrize(
+    'event,raises',
+    [
+        (
+            '[{"records": [{"asset_id": 0}], "metadata": {"app_stream_id": 0, "apps": '
+            '{"%s": {"app_connection_id": 0}}}}]',
+            True,
+        ),
+        (
+            '[{"records": [{"asset_id": 0, "timestamp": 0, "measured_depth": 0}], "metadata": '
+            '{"app_stream_id": 0, "apps": {"%s": {"app_connection_id": 0}}}}]',
+            True,
+        ),
+        (
+            '[{"records": [{"asset_id": 0, "measured_depth": 0}], "metadata": {"app_stream_id": 0, "apps": '
+            '{"%s": {"app_connection_id": 0}}}}]',
+            False,
+        ),
+        (
+            '[{"records": [{"asset_id": 0, "timestamp": 0}], "metadata": {"app_stream_id": 0, "apps": '
+            '{"%s": {"app_connection_id": 0}}}}]',
+            False,
+        ),
+    ],
+    ids=[
+        'no timestamp and no measured_depth provided exc',
+        'both timestamp and measured_depth provided exc ',
+        'only measured_depth provided (correct event)',
+        'only timestamp provided (correct event)',
+    ],
+)
+def test_require_timestamp_or_measured_depth(event, raises):
+    event %= SETTINGS.APP_KEY
+
+    corva = Corva(SimpleNamespace(client_context=None))
+
+    if raises:
+        exc = pytest.raises(ValueError, corva.stream, stream_app, event)
+        assert 'Either timestamp or measured_depth is required' in str(exc.value)
+        return
+
+    corva.stream(stream_app, event)
+
+
+@pytest.mark.parametrize(
+    'event,raises,format_event',
+    [
+        (
+            '[{"records": [{"asset_id": 0, "timestamp": 0}], "metadata": {"app_stream_id": 0, "apps": {}}}]',
+            True,
+            False,
+        ),
+        (
+            '[{"records": [{"asset_id": 0, "timestamp": 0}], "metadata": {"app_stream_id": 0, "apps": '
+            '{"%s": {"app_connection_id": 0}}}}]',
+            False,
+            True,
+        ),
+    ],
+    ids=['no app key exc', 'correct event'],
+)
+def test_require_app_key_in_metadata_apps(event, raises, format_event):
+    if format_event:
+        event %= SETTINGS.APP_KEY
+
+    corva = Corva(SimpleNamespace(client_context=None))
+
+    if raises:
+        exc = pytest.raises(ValueError, corva.stream, stream_app, event)
+        assert 'metadata.apps dict must contain an app key.' in str(exc.value)
```

Comment:
it is not very good to check exception messages. Messages can change, without any changes in logic. Do we really want to debug/test change whenever we change the message?

Reply:
I see, but how to be sure, that the exception is the expected one?

Reply:
I disagree with this proposed change. In this case, we are checking a generic exception that has a specific message, so we need to ensure that we are catching the correct exception.

An alternate approach, if possible, would be to raise a more specific exception.